service/src/prover_service.rs

```diff
@@ -122,24 +128,59 @@ impl ProverService for ProverServiceSVC {
                                     error_message: e.to_string(),
                                 })),
                             }),
-                        RequestType::GenBatchProof(req) => handler_clone
-                            .handle_gen_batch_proof_request(
-                                request_id.clone(),
-                                req,
-                                eth_client_clone.clone(),
-                            )
-                            .await
-                            .unwrap_or_else(|e| ProverResponse {
-                                id: request_id.clone(),
-                                response_type: Some(ResponseType::GenBatchProof(
-                                    GenBatchProofResponse {
-                                        batch_id: "".to_string(),
-                                        result_code: ProofResultCode::CompletedError as i32,
-                                        batch_proof_result: None,
-                                        error_message: e.to_string(),
-                                    },
-                                )),
-                            }),
+                        RequestType::GenBatchProof(req) => {
+                            if let Some(batch_req) = req.step {
+                                match batch_req {
+                                    Step::GenBatchChunks(gen_chunk_req) => {
+                                        handler_clone
+                                            .handle_gen_batch_chunks_request(
+                                                request_id.clone(), gen_chunk_req, eth_client_clone.clone()
+                                            ).await.unwrap_or_else(|e| ProverResponse {
+                                            id: request_id.clone(),
+                                            response_type: Some(ResponseType::GenBatchProof(
+                                                GenBatchProofResponse {
+                                                    step: Some(gen_batch_proof_response::Step::GenBatchChunks(GenBatchChunksResult{
```

Comment:
GenBatchChunksResult{
result_code: ProofResultCode::CompletedError as i32,
rror_message: e.to_string(),
..Defult::default()
}

Do the same for other error handlers.

Reply:
ok